runtime/lua/vim/filetype.lua

```diff
@@ -635,6 +644,12 @@ local extension = {
   tssop = "tssop",
   tutor = "tutor",
   twig = "twig",
+  ts = function(name, buf)
+    if starts_with(buf, '<?xml') then
+      return 'xml'
+    end
+    return 'typescript'
+  end,
```

Comment:
1. `<?xml` doesn't need to be at the very beginning. According to the pattern in `filetype.vim`, it can be anywhere in the first line so use `string.find` rather than `starts_with`. (this is the case for most of the patterns in `filetype.vim`, so I don't think we actually need `starts_with` as a helper function).

2. The `?` character needs to be escaped with `%` since Lua interprets `?` as a pattern operator.

3. We don't have a standard convention across our Lua code base, but in order to at least stay consistent within this one file, use double quotes.

```suggestion
  ts = function(name, bufnr)
    if string.find(getline(bufnr, 1), "<%?xml") then
      return "xml"
    end
    return "typescript"
  end,
```

I'm being a bit picky because we want to set a standard for future contributions, and most people will just follow whatever conventions are established in the file. So I want to make sure we start off right.